polyfill/lib/duration.mjs

```diff
@@ -138,6 +138,9 @@ export class Duration {
   static from(...args) {
     return ES.CastDuration(...args);
   }
+  static fromNanoSeconds(arg) {
+    return new Duration(0, 0, 0, 0, 0, 0, 0, 0, arg);
```

Comment:
Probably we should call new this (to support subclassing)

Reply:
Maybe we should hold off on doing that and make sure we think through all the situations. For example, we can support things like that here, but not so much in things like `withDate()`

Reply:
It seems like a problem if this proposal’s static factory methods don’t support subclassing; i think it’d need a really strong argument for any of them not to.

Reply:
Considering subclasses could have extra constructor arguments (at any position) and there is no way how to initialize them and the constructor can only be called with new, I don’t see how these statics could ever produce anything but the actual intrinsic parent.

@ljharb please enlighten me, this seems to be a total non issue and calling new this() could be entirely bad!